### PR TITLE
Rd 3457 GitHub scraping

### PR DESCRIPTION
**Pull request by glukhman**

### CODE REVIEW

File: plugins/alembic/env.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 from sqlalchemy import pool
 from sqlalchemy.ext.asyncio import AsyncEngine
 
-from plugins.app.models import Base
+from app.models import Base
```
Comment (mateumann) [inline plugins/alembic/env.py:9]: why this change?

File: scraping-tool/scrape_plugins.py [hunk 2]
```diff
@@ -23,51 +25,94 @@ def main(args):
             auth=(args.github_user, args.github_token))
         if not versions or versions.status_code != 200:
             continue
```
Comment (mateumann) [inline scraping-tool/scrape_plugins.py:27]:
```suggestion
            logger.warning(f"Didn't process {plugin['repo']}: {response.text}")
            continue
```

File: scraping-tool/scrape_plugins.py [hunk 3]
```diff
@@ -23,51 +25,94 @@ def main(args):
...
+            if plugin_version.status_code != 201:
+                raise Exception("{} {}".format(
+                    plugin_version.status_code, plugin_version.text))
+
+            version_id = plugin_version.text
+            logger.info('Created plugin version entry %s', version_id)
+
+            # TODO: scrape node types
+
+        #
```
Comment (mateumann) [inline scraping-tool/scrape_plugins.py:96]: maybe remove that commented lines (for clarity)?

File: scraping-tool/scrape_plugins.py [hunk 4]
```diff
@@ -79,19 +124,19 @@ def _append_wagons_list(file, wagons):
                 distro = distro.split('.')[0]
                 if distro == "Centos Core" and 'aarch64' in file:
```
Comment (mateumann) [inline scraping-tool/scrape_plugins.py:125]: It's not pretty.  Maybe we'd better add one more column to the DB (arch/architecture)?

File: scraping-tool/scrape_plugins.py [hunk 5]
```diff
@@ -23,52 +24,76 @@ def main(args):
             auth=(args.github_user, args.github_token))
         if not versions or versions.status_code != 200:
             continue
```
Comment (mateumann) [inline scraping-tool/scrape_plugins.py:26]: Why not warn user about inability to fetch `versions`?

File: scraping-tool/scrape_plugins.py [hunk 6]
```diff
@@ -23,52 +24,76 @@ def main(args):
...
+                continue
+                # TODO :: if param=replace_existing, delete version
 
+            # create plugin version entry
             version_data = \
                 requests.get('{0}/repos/{1}/releases/tags/{2}'.format(
                     GB_API_URL_PRE, plugin['repo'], version_name),
                     auth=(args.github_user, args.github_token))
             if version_data and version_data.status_code == 200:
                 p_yaml, wagons = get_plugin_yaml_and_wagons(version_data)
```
Comment (mateumann) [inline scraping-tool/scrape_plugins.py:73]: warn otherwise?

File: scraping-tool/scrape_plugins.py [hunk 7]
```diff
@@ -77,21 +102,25 @@ def _append_wagons_list(file, wagons):
                 distro = "{} {}".format(tag.capitalize(),
                                         file.split(tag)[1].split('-')[1])
```
[the remark this answers is not in the document]
Comment (glukhman) [reply scraping-tool/scrape_plugins.py:104]: This pattern is what the wagon packager produces. The other option is also matching for distro sub-versions, which I don't like

**Review by mateumann — APPROVED** [review]
